src/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+target_sources(Template_Hello PRIVATE
+		main.cpp
+)
+target_sources(Template_Template PRIVATE
+		template.cpp
+)
+
+target_link_libraries(Template_Template PRIVATE fmt::fmt)
+target_link_libraries(Template_Hello PRIVATE Template_Template)
+set_target_properties(Template_Template PROPERTIES
+		PUBLIC_HEADER template.h
+)
```

Comment:
One thing that will be tricky is designing for both `PUBLIC_HEADER` and `target_source(FILE_SET)`. I haven't got around to figuring out how to do that.